src/rgw/rgw_civetweb_frontend.cc

```diff
@@ -65,8 +63,13 @@ int RGWCivetWebFrontend::run()
   }
 
   /* Prepare options for CivetWeb. */
-  const std::set<boost::string_ref> rgw_opts = { "port", "prefix" };
-  const size_t CW_NUM_OPTS = 2 * (conf_map.size() - rgw_opts.size()) + 1;
+  const std::set<string> rgw_opts = { "port", "prefix" };
+  size_t rgw_opt_num = 0;
+  for (const auto& opt: rgw_opts){
+    if (conf_map.find(opt) != conf_map.end())
+      rgw_opt_num++;
+  }
+  const size_t CW_NUM_OPTS = 2 * (conf_map.size() - rgw_opt_num) + 1;
   const char *options[CW_NUM_OPTS];
```

Comment:
I'm very surprised this is compiling-- I'd expect at least a warning. C++ doesn't support variable-length arrays, and conf_map isn't a constant...